clients/unshaded/src/main/java/tachyon/client/BlockMasterClient.java

```diff
@@ -71,7 +72,7 @@ protected void afterConnect() {
    * @return A list of worker info returned by master
    * @throws IOException if an I/O error occurs
    */
-  public synchronized List<WorkerInfo> getWorkerInfoList() throws IOException {
+  public synchronized List<WorkerInfo> getWorkerInfoList() throws IOException, TachyonException {
```

Comment:
Could we declare all these methods to throw ConnectionFailedException instead? That would make it clearer what is being handled